components/heap/third_party/dartino/cmpctmalloc.c

```diff
@@ -217,11 +217,16 @@ typedef struct free_struct {
     struct free_struct *prev;
 } free_t;
 
+typedef enum {
+  PAGE_FREE = 0,
+  PAGE_IN_USE = 1,    // Page is first in an allocation.
+  PAGE_CONTINUED = 2  // Page is subsequent in an allocation.
+} page_use_t;
+
 // For page allocator, not originally part of cmpctmalloc.  These fields are 32 bit
 // so that they work in IRAM on ESP32.
 typedef struct Page {
-    uint32_t continued;  // Boolean - this is part 2 or more of a series of pages.
-    uint32_t in_use;     // Boolean.
+    uint32_t status;
```

Comment:
We need 2 bits to store the status, and we can't use char because it might be IRAM, but we could pack 16 in a 32 bit int.  Right now the overhead is 8 bytes per page. We could reduce it to 4.25 bytes per page, but the code would be a lot more complicated, so I don't think it's worth it.